drivers/spi/spi_mcux_flexcomm.c

```diff
@@ -719,6 +736,10 @@ static int transceive(const struct device *dev,
 	struct spi_mcux_data *data = dev->data;
 	int ret;
 
+	#ifdef CONFIG_PM_POLICY_DEVICE_CONSTRAINTS
+	pm_policy_device_power_lock_get(dev);
+	#endif
```

Comment:
I'm not sure if the intention is for this to be peppered all over in every driver's code, or what the future of this API is.  @bjarki-andreasen can you please weigh in on this

Reply:
I actually fix this as part of the power domain PR I'm working on, there is a bug in the policy.h header :) See https://github.com/zephyrproject-rtos/zephyr/pull/89372/commits/97589c473884500501da953becf76370ccce37e6 

@LuisC-NXP you can cherry pick the above commit, this removes the need for the ifdefs :)